### PR TITLE
add files for mobile app

### PR DESCRIPTION
**Pull request by rumanod**
https://jira.2u.com/browse/CTED-60

### CODE REVIEW

File: appjs/app_init.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+(function (t) {
+
+    /* Register a link handler to open mod/groupselect/view.php links anywhere in the app. */
+    function AddonModGroupselectModuleLinkHandler() {
+        t.CoreContentLinksModuleIndexHandler.call(this, t.CoreCourseHelperProvider, 'mmaModGroupselect', 'groupselect');
+
+        this.pattern = /\/mod\/groupselect\/view\.php.*([\&\?]id=\d+)/;
+        this.name = "AddonModGroupselectLinkHandler";
+
+        console.log(this.pattern);
```
Comment (JJSwanevelder) [inline appjs/app_init.js]: Please remove this

File: appjs/app_init.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    AddonModGroupselectModuleLinkHandler.prototype = Object.create(t.CoreContentLinksModuleIndexHandler.prototype);
+    AddonModGroupselectModuleLinkHandler.prototype.constructor = AddonModGroupselectModuleLinkHandler;
+
+    AddonModGroupselectModuleLinkHandler.prototype.getActions = function (siteIds, url, params, courseId) {
+        console.log('PP2');
```
Comment (JJSwanevelder) [inline appjs/app_init.js]: Please remove

File: classes/output/mobile.php [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        $groupmode = groups_get_activity_groupmode ( $cm, $course );
+        $counts = groupselect_group_member_counts ( $cm, $groupselect->targetgrouping );
+        $groups = groups_get_all_groups ( $course->id, 0, $groupselect->targetgrouping );
+        $passwordgroups = groupselect_get_password_protected_groups ( $groupselect );
+        $hidefullgroups = $groupselect->hidefullgroups;
+        $exporturl = '';
+        $ASSIGNROLE = 4;
+
+        // Permissions
+        $accessall = has_capability ( 'moodle/site:accessallgroups', $context );
```
Comment (JJSwanevelder) [inline classes/output/mobile.php:64]: Not sure if we should have so many open spaces and for the few lines below the same

File: appjs/app_init.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            action: function() {
+                let site = t.CoreSitesProvider.getCurrentSite();
+                let navigation = site.appProvider.getRootNavController();
+
+                const pageParams = {
+                    cmid: parseInt(params.id, 10),
+                };
+
+                return Promise.resolve(t.CoreSitePluginsProvider.getContent('mod_groupselect', 'groupselect_view', pageParams)).then((contentResult) => {
+                    console.log('PP1');
```
Comment (PPrinsloo) [inline appjs/app_init.js:48]: Please remove.

File: classes/output/mobile.php [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+            $grpname = format_string ( $group->name, true, array (
+                    'context' => $context
+            ) );
+
+            // Skips listing full groups if set
+            if (! $ismember and $hidefullgroups and $groupselect->maxmembers === $usercount) {
+                continue;
+            }
+
+            if (in_array ( $group->id, $passwordgroups )) {
```
Comment (JJSwanevelder) [inline classes/output/mobile.php:100]: Can use turnerary statement if it fits in reasonable line length.

File: classes/output/mobile.php [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+            }
+
+            // Group members
+            if ($canseemembers) {
+                if ($members = groups_get_members ( $group->id )) {
+                    $membernames = array ();
+                    foreach ( $members as $member ) {
+                        $pic = $OUTPUT->user_picture ( $member, array (
+                                'courseid' => $course->id
+                        ) );
```
Comment (JJSwanevelder) [inline classes/output/mobile.php:135]: Weird spacing here.

**Review by PPrinsloo — CHANGES_REQUESTED** [review]

File: classes/output/mobile.php [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+                    ) ), get_string ( 'select', 'mod_groupselect', "") );
+                    $actionpresent = true;
+                } else {
+                    $line->maxlimitreached = '';
+                }
+            }
+                $data [] = $line;
+        }
+
+        //$data = json_decode (json_encode ($data), FALSE);
```
Comment (JJSwanevelder) [inline classes/output/mobile.php:213]: Remove please

File: templates/mobile_view_add_new_group.mustache [hunk 8]
```diff
@@ -0,0 +1 @@
+{{=<% %>=}}
```
Comment (JJSwanevelder) [inline templates/mobile_view_add_new_group.mustache:1]: Not sure what is the reason for this file if only this line is here.

File: templates/mobile_view_groupselect.mustache [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+            [args]="{cmid: <%cmid%>}" 
+            title="Create New Group">
+            Create New Group
+        </button>
+    </ion-card>
+
+<ion-card *ngFor="let group of CONTENT_OTHERDATA.groups"
+    core-site-plugins-new-content >
+    <!-- Header -->
+    <ion-item style="background-color: #f7f7f7;">
```
Comment (JJSwanevelder) [inline templates/mobile_view_groupselect.mustache:18]: If possible do all styling in the stylesheets. We should only continue with this styling if we dynamically bind angular styles. Even so we can dynamically bind classes. This is also for all other inline styling below.

**Review by JJSwanevelder — CHANGES_REQUESTED** [review]
Hey @rumanod please see suggested changes.

**Review by JJSwanevelder — APPROVED** [review]
👍

**Pull request merged**